test/distributed/test_nvshmem_triton.py

```diff
@@ -413,50 +412,54 @@ def test_triton_put_signal_add(self) -> None:
                 flag, torch.tensor([SIGNAL_VAL], dtype=torch.int64, device=self.device)
             )
 
-    # This test hangs. TODO: investigate why.
-    @skip_but_pass_in_sandcastle("Hangs")
     @skipIfRocm
     @requires_triton()
     def test_triton_wait_until(self) -> None:
         torch.manual_seed(42 + self.rank)
         self._init_device()
+
         nvshmem_lib = nvshmem.enable_triton()
         group_name = dist.group.WORLD.group_name
         symm_mem.enable_symm_mem_for_group(group_name)
+
         rank = self.rank
+        peer = 1 - rank
```

Comment:
hmm I know the world size is 2 and we write it in this test already. Really not a big fan of it.. maybe (self.world_size - 1) - rank?